### PR TITLE
Build: always use `gvisor` Docker runtime

### PR DESCRIPTION
**Pull request by humitos**
Remove the feature flag to decide whether or not Docker should use `gVisor`. This is currently enabled by default to all the projects since last week, so I think we are safe to remove it since it has been working fine.

Related #9779

### CODE REVIEW

File: readthedocs/doc_builder/environments.py
Comment (agjohnson) [inline readthedocs/doc_builder/environments.py:816]: I wonder if this should be an application setting, we don't strictly need gVisor for local development. Otherwise, we should update our install docs or something to mention the additional system dependency on gVisor. Having parity with prod is always nice though.

**Review by agjohnson — APPROVED** [review]

File: readthedocs/doc_builder/environments.py
Comment (humitos) [reply readthedocs/doc_builder/environments.py:816]: We have this guide https://dev.readthedocs.io/en/latest/guides/gvisor.html

Maybe we need to merge it / mention it from the development installation instructions.

I prefer if we use gVisor in development, for parity as well. I've been using it and I had no problems.

Comment (agjohnson) [reply readthedocs/doc_builder/environments.py:816]: Yeah, linking to that in the install procedure is enough. I'm pretty sure we all went through this install already even, so it won't be a breaking change to core team.

**Comment by humitos** [comment]
This is ready to be merged, but we will wait a couple of weeks before doing so. We enabled gVisor for all projects some days ago. I'll come back to this PR after these weeks and merge it.

**Comment by humitos** [comment]
I'm merging. This is going out next week.

**Pull request merged**
